homeassistant/components/niko_home_control/__init__.py

```diff
@@ -50,34 +51,14 @@ async def async_unload_entry(
     return await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
 
 
-class NikoHomeControlData:
-    """The class for handling data retrieval."""
-
-    def __init__(self, hass, nhc):
-        """Set up Niko Home Control Data object."""
-        self.nhc = nhc
-        self.hass = hass
-        self.available = True
-        self.data = {}
-        self._system_info = None
-
-    @Throttle(MIN_TIME_BETWEEN_UPDATES)
-    async def async_update(self):
-        """Get the latest data from the NikoHomeControl API."""
-        _LOGGER.debug("Fetching async state in bulk")
-        try:
-            self.data = await self.hass.async_add_executor_job(
-                self.nhc.list_actions_raw
-            )
-            self.available = True
-        except OSError as ex:
-            _LOGGER.error("Unable to retrieve data from Niko, %s", str(ex))
-            self.available = False
-
-    def get_state(self, aid):
-        """Find and filter state based on action id."""
-        for state in self.data:
-            if state["id"] == aid:
-                return state["value1"]
-        _LOGGER.error("Failed to retrieve state off unknown light")
-        return None
+class NHCController(_NHCController):
+    """The niko home control controller."""
+
+    def __init__(self, host, port) -> None:
+        """Init niko home control controller."""
+        super().__init__(host, port)
+        self.entities: dict[str, Any] = {}
+
+    def get_entity(self, action_id):
+        """Get entity by id."""
+        return self.entities[action_id]
```

Comment:
We should not keep track of created entities outside of the entity registries. Rather allow the library to accept multiple callbacks and register them in the entity or use a dispatcher mechanism

Reply:
resolved (I think)